protocol/otel-protocol.cabal

```diff
@@ -0,0 +1,49 @@
+cabal-version: 3.6
```

Comment:
Making the generated files into separate packages is not needed, and I believe it even increases deployment complexity.
Also, please keep the files in the `generated` directory to indicate that they were generated.

Reply:
The main reason for moving the generated files into a separate package was to just use the files obtained with `protoc`, without touching them. Having a different package would simplify the configuration of _cabal_ options like the language and the default ghc extensions. We could try to add a second library in the existing `otel-haskell-sdk.cabal` file.

Reply:
It works using an internal library, what do you think?

Reply:
Looks ok :)